addon/webextension/background/analytics.js

```diff
@@ -0,0 +1,74 @@
+/* globals main, auth, catcher, deviceInfo, communication */
+
+window.analytics = (function () {
+  let exports = {};
+
+  let telemetryPrefKnown = false;
+  let telemetryPref;
+
+  exports.sendEvent = function (action, label, options) {
+    let eventCategory = "addon";
+    if (! telemetryPrefKnown) {
+      console.warn("sendEvent called before we were able to refresh");
+      return Promise.resolve();
+    }
+    if (! telemetryPref) {
+      console.info(`Cancelled sendEvent ${eventCategory}/${action}/${label || 'none'} ${JSON.stringify(options)}`);
```

Comment:
Seems like [`window.dump()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/dump) would be a good alternative to pref-setting here.